src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlSessionFactory.java

```diff
@@ -0,0 +1,157 @@
+package io.stargate.sgv2.jsonapi.service.cqldriver;
+
+import static io.stargate.sgv2.jsonapi.util.ClassUtils.classSimpleName;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.CqlSessionBuilder;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.metadata.schema.SchemaChangeListener;
+import io.stargate.sgv2.jsonapi.config.DatabaseType;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.optvector.SubtypeOnlyFloatVectorToArrayCodec;
+import java.net.InetSocketAddress;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Factory to create {@link CqlSession} instances, normally used with the {@link CQLSessionCache}
+ * via the {@link CQLSessionCache.SessionFactory} interface.
+ *
+ * <p>Abstracted out to make it easier to test the session cache and creating the session.
+ */
+public class CqlSessionFactory implements CQLSessionCache.SessionFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CqlSessionFactory.class);
+
+  private final String applicationName;
+
+  private final DatabaseType databaseType;
+  private final String localDatacenter;
+  private final Collection<InetSocketAddress> contactPoints;
+  private final List<SchemaChangeListener> schemaChangeListeners;
+  private final Supplier<CqlSessionBuilder> sessionBuilderSupplier;
+
+  CqlSessionFactory(
```

Comment:
Should perhaps add brief Javadoc mentioning this is the main non-test Constructor to use?
(since the other with `Supplier<CqlSessionBuilder>` is for use by test as per comments)